shell/platform/android/test/io/flutter/embedding/android/FlutterAndroidComponentTest.java

```diff
@@ -0,0 +1,186 @@
+package io.flutter.embedding.android;
+
+import android.app.Activity;
+import android.arch.lifecycle.Lifecycle;
+import android.content.Context;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.robolectric.Robolectric;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.annotation.Config;
+
+import io.flutter.embedding.engine.FlutterEngine;
+import io.flutter.embedding.engine.FlutterEngineCache;
+import io.flutter.embedding.engine.FlutterJNI;
+import io.flutter.embedding.engine.FlutterShellArgs;
+import io.flutter.embedding.engine.loader.FlutterLoader;
+import io.flutter.embedding.engine.plugins.FlutterPlugin;
+import io.flutter.embedding.engine.plugins.activity.ActivityAware;
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
+
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+
+@Config(manifest=Config.NONE)
+@RunWith(RobolectricTestRunner.class)
+public class FlutterAndroidComponentTest {
+  @Test
+  public void pluginsReceiveFlutterPluginBinding() {
+    // ---- Test setup ----
+    // Place a FlutterEngine in the static cache.
+    FlutterLoader mockFlutterLoader = mock(FlutterLoader.class);
+    FlutterJNI mockFlutterJni = mock(FlutterJNI.class);
+    when(mockFlutterJni.isAttached()).thenReturn(true);
+    FlutterEngine cachedEngine = spy(new FlutterEngine(RuntimeEnvironment.application, mockFlutterLoader, mockFlutterJni));
+    FlutterEngineCache.getInstance().put("my_flutter_engine", cachedEngine);
+
+    // Add mock plugin.
+    FlutterPlugin mockPlugin = mock(FlutterPlugin.class);
+    cachedEngine.getPlugins().add(mockPlugin);
+
+    // Create a fake Host, which is required by the delegate.
+    FlutterActivityAndFragmentDelegate.Host mockHost = mock(FlutterActivityAndFragmentDelegate.Host.class);
```

Comment:
readability nit: It would probably be better to create a stub than a mock here. Subjective but stubs are generally easier to reason about than a series of `when...thenReturn()` statements.

Reply:
Sure. I switched out those mocks for fake implementations.